client/extensions/woocommerce/state/sites/reviews/handlers.js

```diff
@@ -62,3 +79,99 @@ export function handleReviewsRequestError( { dispatch }, action, error ) {
+	const { siteId, reviewId, newStatus } = action;
+	// The product reviews endpoint batch/update endpoints seem busted in v3, v2, and v1
+	// We can use the comments endpoint for now. One difference is pending vs hold
+	const statusToPass = 'pending' === newStatus ? 'hold' : newStatus;
+	dispatch( request( siteId, action, '/wp/v2' ).post( `comments/${ reviewId }`, { status: statusToPass } ) );
+}
+
+export function handleChangeReviewStatusSuccess( { dispatch, getState }, action ) {
+	const state = getState();
+	const { siteId, currentStatus, newStatus } = action;
+
+	// Refreshes the review list if dealing with trash and spam.
+	// otherwise pending/approved reviews stay in their current lists.
+	// This matches the behavior of comments management.
+	if ( 'spam' === currentStatus || 'spam' === newStatus || 'trash' === currentStatus || 'trash' === newStatus ) {
+		const currentPage = getReviewsCurrentPage( state, siteId );
+		const currentSearch = getReviewsCurrentSearch( state, siteId );
+		const query = {
+			page: currentPage,
+			search: currentSearch,
+			status: currentStatus,
+		};
+
+		if ( '' !== currentSearch ) {
+			query.status = 'any';
+		}
+
+		dispatch( fetchReviews( siteId, query ) );
+	}
+
+	const message = {
+		approved: translate( 'Review approved.' ),
+		pending: translate( 'Review unapproved.' ),
+		spam: translate( 'Review marked as spam.' ),
+		trash: translate( 'Review trashed.' ),
+	};
+	const defaultMessage = translate( 'Review status updated' );
+
+	dispatch( successNotice( get( message, newStatus, defaultMessage ), {
+		duration: 5000,
+	} ) );
+}
+
+export function announceStatusChangeFailure( { dispatch }, action ) {
+	const { reviewId, currentStatus } = action;
+
+	// Reverts the local status back to what it was if the remote request failed.
+	dispatch(
+		bypassDataLayer( {
+			...omit( action, [ 'meta' ] ),
+			newStatus: currentStatus,
+		} )
+	);
+
+	const errorMessage = {
+		approved: translate( "We couldn't approve this review." ),
+		pending: translate( "We couldn't unapprove this review." ),
+		spam: translate( "We couldn't mark this review as spam." ),
+		trash: translate( "We couldn't move this review to trash." ),
+	};
+	const defaultErrorMessage = translate( "We couldn't update this review." );
+
+	dispatch( errorNotice( get( errorMessage, currentStatus, defaultErrorMessage ), {
+		id: `review-notice-error-${ reviewId }`,
+	} ) );
+}
+
+export function handleDeleteReview( { dispatch }, action ) {
+	const { siteId, reviewId } = action;
+	dispatch( request( siteId, action, '/wp/v2' ).del( `comments/${ reviewId }?force=true` ) );
+}
+
+export function announceDeleteSuccess( { dispatch, getState }, action ) {
+	const state = getState();
+	const { siteId } = action;
+	const currentPage = getReviewsCurrentPage( state, siteId );
+	dispatch( fetchReviews( siteId, { search: '', page: currentPage, status: 'trash' } ) );
+
+	dispatch(
+		successNotice(
+			translate( 'Review deleted permanently.' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Comment deleted permanently.' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).